compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/MoveCompanionObjectFieldsLowering.kt

```diff
@@ -83,6 +86,16 @@ private class MoveCompanionObjectFieldsLowering(val context: CommonBackendContex
         companion.declarations.removeAll { it is IrAnonymousInitializer }
     }
 
+    private fun copyConsts(irClass: IrClass) {
+        val companion = irClass.declarations.find {
+            it is IrClass && it.isCompanion
+        } as IrClass? ?: return
+        val unused = mutableMapOf<IrFieldSymbol, IrFieldSymbol>()
+        companion.declarations.filter { it is IrProperty && it.isConst }.mapNotNullTo(irClass.declarations) {
+            movePropertyFieldToStaticParent(it as IrProperty, companion, irClass, unused)
```

Comment:
In such cases old field should not be replaced, all referenced should be kept. New copy is not used in Kotlin, it's created only for Java

Reply:
@max-kammerer Thanks for the explanation. I misunderstood how the old field is held.

In the case of copy, the latest patch simply doesn't touch `irProperty.backingField`, so that the old field is retained. The new field is held directly by the interface and doesn't have a `correspondingPropertySymbol`. The references are not updated, of course. Does that make sense?

Please take a look again. I also verified with an example and saw that both new and old const exist.

Reply:
Yes, everything is fine except obsolete name and desc